cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/GoalUtilsTest.java

```diff
@@ -0,0 +1,91 @@
+                                                       resource,
+                                                       ResourceDistributionGoal.BALANCE_MARGIN,
+                                                       false,
+                                                       true,
+                                                       0.4,
+                                                       1.3,
+                                                       1.2,
+                                                       0.4 * ResourceDistributionGoal.BALANCE_MARGIN);
+
+    // Verify 3: Not low utilization and compute balance threshold lower bound
+    verifyComputingResourceUtilizationBalanceThreshold(0.3,
+                                                       resource,
+                                                       ResourceDistributionGoal.BALANCE_MARGIN,
+                                                       true,
+                                                       true,
+                                                       0.2,
+                                                       1.3,
+                                                       1.2,
+                                                       0.3 * (1 - ((1.3 * 1.2) - 1) * ResourceDistributionGoal.BALANCE_MARGIN));
+
+    // Verify 4: Not low utilization and compute balance threshold upper bound
+    verifyComputingResourceUtilizationBalanceThreshold(0.3,
+                                                       resource,
+                                                       ResourceDistributionGoal.BALANCE_MARGIN,
+                                                       false,
+                                                       true,
+                                                       0.2,
+                                                       1.3,
+                                                       1.2,
+                                                       0.3 * (1 + ((1.3 * 1.2) - 1) * ResourceDistributionGoal.BALANCE_MARGIN));
+
+  }
+
+  private void verifyComputingResourceUtilizationBalanceThreshold(double avgUtilizationPercentage,
+                                                                  Resource resource,
+                                                                  double balanceMargin,
+                                                                  boolean isLowerThreshold,
+                                                                  boolean isTriggeredByGoalViolation,
+                                                                  double lowUtilizationThreshold,
+                                                                  double resourceBalancePercentage,
+                                                                  double goalViolationDistributionThresholdMultiplier,
+                                                                  double expectedComputedBalanceThreshold) {
```

Comment:
We have a lot of parameters, but seems like we only test a few. Can we either drop the constant parameters and assign the value within this function or add test cases that varies the tested parameter.

Reply:
I will `drop the constant parameters and assign the value within this function`